src/olympia/accounts/views.py

```diff
@@ -385,12 +390,36 @@ def logout_user(request, response):
         API_TOKEN_COOKIE, domain=settings.SESSION_COOKIE_DOMAIN)
 
 
+# This view is not covered by the CORS middleware, see:
+# https://github.com/mozilla/addons-server/issues/11100
 class SessionView(APIView):
     permission_classes = [IsAuthenticated]
 
+    def add_common_cors_headers(self, request, response):
+        origin = request.META.get('HTTP_ORIGIN')
```

Comment:
if `origin` is not set, we probably don't want to add the headers, and return straight away?

Reply:
ah yes, that would make sense.

Reply:
Don't forget to do something sensible for `OPTIONS` too (not sure what)

Reply:
Looks like the middleware does not add anything when `Origin` is missing.